bot/util/utilities.py

```diff
@@ -198,6 +268,64 @@ def generate_features_string(features):
     return str_features
 
 
+def get_text_channel(channel_id: str, guild: discord.Guild) -> discord.TextChannel:
+    """Parses a message id string and searches the text channel in the passed guild
+
+    Args:
+        channel_id (str): The id of the channel (might also be surrounded by '<#' and '>'
+        guild (discord.Guild): The guild in which the channel is searched.
+
+    Returns:
+        discord.Channel: The found channel.
+
+    Raises:
+        ValueError: If the chanel could not be found.
+    """
+    try:
+        if channel_id.startswith('<#'):
+            channel_id = channel_id[2:-1]
+        channel = guild.get_channel(int(channel_id))
+    except ValueError:
+        raise ValueError('Channel to post embed to could not be found. Use channel ID or channel name (linked #)')
```

Comment:
you cannot type a valid channel name in discord without it becoming automatically a mention. So I took searching by name and by id for the same

Reply:
When I hear searching by name, I think of "test-channel" and not "#test-channel". So this functions is searching by channel ID or mention.

Reply:
Technically it still is the channel name, as all channel names start with #. The answer returned to the user also states again to use #

Reply:
I get your idea but for Discord something like "#test-channel" is a mention because it resolves to a channel id and that's why I would write something like `[...] Use valid channel ID or a mention instead`. It just isn't the same in my opinion.

Reply:
will change